NeoML/Python/neoml/Dnn/Loss.py

```diff
@@ -741,15 +741,66 @@ def __init__(self, input_layers, loss_weight=1.0, name=None):
 
 
 class CustomLossCalculatorBase(metaclass=ABCMeta):
-    """
+    """The base class that you should implement to calculate the custom loss function.
     """
     @abstractmethod
     def calc(self, data, labels):
-        """
+        """Calculates the custom loss function.
+        This function may use only the operations supported for autodiff:
+
+        - simple arithmetic: `+ - * /`
+        - the `neoml.AutoDiff.*` functions
+        - `neoml.Autodiff.const` for creating additional blobs filled with given values
+
+        :param neoml.Blob.Blob data: the network response.
+
+        :param neoml.Blob.Blob labels: the correct labels.
         """
 
 class CustomLoss(Loss):
-    """
+    """The layer that calculates a custom loss function.
+
+    :param input_layers: the input layers to be connected. 
+        The integer in each tuple specifies the number of the output.
+        If not set, the first output will be used.
+    :type input_layers: list of object, tuple(object, int)
+    :param neoml.Dnn.CustomLossCalculatorBase loss_calculator: a user-implemented object 
+        that provides the method to calculate the custom loss.
+    :param loss_weight: the multiplier for the loss function value during training.
+    :type loss_weight: float, default=1.0
+    :param name: the layer name.
+    :type name: str, default=None
+
+    .. rubric:: Layer inputs:
+
+    (1) the network response for which you are calculating the loss.
+        It should contain the probability distribution for objects over classes.
+        If you are not going to apply softmax in this layer, each element should already be >= 0, 
+        and the sum over **Height** * **Width** * **Depth** * **Channels** dimension should be equal to 1.
+        The dimensions:
+
+        - **BatchLength** * **BatchWidth** * **ListSize** - the number of objects
+        - **Height** * **Width** * **Depth** * **Channels** - the number of classes
+    
+    (2) the correct class labels. Two formats are acceptable:
```

Comment:
At this moment only the former of these formats is supported in custom loss.

Reply:
Fixed.